text/0000-rustdoc-json.md

```diff
@@ -0,0 +1,1124 @@
+`extern_crates` | Map<int, [ExternalCrate](#ExternalCrate)> | A map of "crate numbers" to metadata about that crate.
+
+### Resolving IDs
+
+The crate's `index` contains mostly local items, which includes impls of external traits on local
+types or local traits on external types. The exception to this is that external trait definitions
+and their associated items are also included in the `index` because this information is useful when
+generating the comprehensive list of methods for a type.
+
+This means that many IDs aren't included in the `index` (any reference to a struct, macro, etc.
+from a different crate). In these cases the fallback is to look up the ID in the crate's `paths`.
+That gives [enough information](#ItemSummary) about the item to create cross references or simply
+provide a name without copying all of the information about external items into the local
+crate's JSON output.
+
+### ExternalCrate
+
+Name      | Type    | Description
+----------|---------|------------------------------------------------------------------------------
+`name`    | String  | The name of the crate.
+`html_root_url` | String  | (*Optional*) The `html_root_url` for that crate if they specify one.
+
+### ItemSummary
+
+Name      | Type    | Description
+----------|---------|------------------------------------------------------------------------------
+`crate_num` | int   | A number corresponding to the crate this Item is from. Used as an key to the `extern_crates` map in [Crate](#Crate). A value of zero represents an Item from the local crate, any other number means that this Item is external.
+`path`    | [String] | The fully qualified path (e.g. `["std", "io", "lazy", "Lazy"]` for `std::io::lazy::Lazy`) of this Item.
+`kind`    | String  | What type of Item this is (see [Item](#Item)).
+
+## Item
+
+An Item represents anything that can hold documentation - modules, structs, enums, functions,
+traits, type aliases, and more. The Item data type holds fields that can apply to any of these,
+and leaves kind-specific details (like function args or enum variants) to the `inner` field.
+
+Name      | Type    | Description
+----------|---------|------------------------------------------------------------------------------
+`crate_num` | int   | A number corresponding to the crate this Item is from. Used as an key to the `extern_crates` map in [Crate](#Crate). A value of zero represents an Item from the local crate, any other number means that this Item is external.
+`name`    | String  | The name of the Item, if present. Some Items, like impl blocks, do not have names.
+`span`    | [Span](#Span) | (*Optional*) The source location of this Item.
+`visibility` | String | `"default"`, `"public"`, `"crate"`, or `"restricted"` (`pub(path)`).
+`restricted_path` | String | (*Optional*) If `visitility == "restricted"`, this field contains the path that it's restricted to.
+`docs`    | String  | The extracted documentation text from the Item.
```

Comment:
Will this include resolved intra-doc links? If not, is it possible to add them somehow?

Reply:
Yes, see [the other comment about docs](https://github.com/rust-lang/rfcs/pull/2963#discussion_r463161567)